src/lib/src/receiver.cpp

```diff
@@ -110,6 +100,8 @@ int roc_receiver_bind(roc_receiver* receiver,
         return -1;
     }
 
+    receiver->addresses.grow(receiver->addresses.size() + 1);
```

Comment:
`addresses.grow()` returns a false if some error occurred. It should be handled here.

Reply:
It also would be great to grow it exponentially instead of linearly.

Reply:
@dshil yes, I will handle it

@gavv ok: I multiply the `receiver->addresses.size()` by 2 when we've reached the `Array.max_size()` for example ?

Reply:
Yep.

Reply:
done